test/e2e/framework/kubectl/kubectl_utils.go

```diff
@@ -142,6 +142,10 @@ func (tk *TestKubeconfig) WriteFileViaContainer(podName, containerName string, p
 		}
 	}
 	command := fmt.Sprintf("echo '%s' > '%s'; sync", contents, path)
+	// TODO: replace with `framework.NodeOSDistroIs` when #81245 is complete
+	if e2epod.NodeOSDistroIs("windows") {
+		command = fmt.Sprintf("echo '%s' > '%s';", contents, path)
```

Comment:
does Windows not have a sync functionality? The reason why we added a sync call is because there were a lot of test flakes due to the data not getting flushed to disk.

Reply:
yeah, I remember discussing this with @jingxu97 and exploring the option of adding https://docs.microsoft.com/en-us/sysinternals/downloads/sync to agnhost, I forgot to do that but you're right, @jingxu97 also mentioned the flakiness and suggested adding the solution in https://github.com/kubernetes/kubernetes/pull/100057

Reply:
there is no easy way for Windows to sync on file (needs extra tools etc.). So we want to give it a try for now. If it flake, we can add logic similar to #100057

Reply:
I don't think that workaround is going to work in all cases, especially for tests that want to test multiple pods sharing the volume on the same node.

Reply:
Can you add a TODO here and open an issue to track adding sync support to the test image?

Reply:
Done, tracked in https://github.com/kubernetes/kubernetes/issues/101172